job_executor/model/datastore.py

```diff
@@ -194,18 +211,20 @@ def delete_draft(self, job_id: str, dataset_name: str):
             self.draft_version.get_dataset_operation(dataset_name)
         )
         if dataset_release_status not in ['PENDING_RELEASE', 'DRAFT']:
```

Comment:
Why not 'PENDING_DELETE' is present in the list? Should not be possible to delete a draft with release status = PENDING_DELETE?

Reply:
Absolutely :100: This needs to be rewritten 👍🏻